webapp/src/main/java/com/cmds/webapp/controllers/APIController.java

```diff
@@ -210,4 +245,31 @@ public Delivery getDeliveryStatus(@PathVariable("id") String id) throws IOExcept
     public String getEndpoint() throws IOException {
         return "test";
     }
+
+    /**
+     * API Call for getting a JSON formatted object of all deliveries for a given Robot ID.
+     * @param id String Robot ID.
+     * @return A Json formatted list of deliveries for a particular robot.
+     * @throws JSONException
+     */
+    @GetMapping("getRobotDeliveries/{id}")
+    public String getRobotDeliveries(@PathVariable("id") String id) throws JSONException  {
+        System.out.println("getRobotDeliveries() API");
+
+        Robot robot = robotRepo.findByName(id);
+        JSONObject robotObject = new JSONObject();
+        if (robot != null) {
+            JSONObject deliveryObject = new JSONObject();
+            for (Delivery d: robot.getListTrips()) {
+                deliveryObject.put("sourceDest", d.getStartingDest());
+                deliveryObject.put("finalDest", d.getFinalDest());
+            }
+            robotObject.put("id", deliveryObject);
```

Comment:
This logic is not correct. You need to move this inside the for loop and have something like:
`robotObject.put(d.getId(), deliveryObject)` this way you get a list of deliveries where the key is the delivery id.

Reply:
Believe this is good now but please confirm